barman/recovery_executor.py

```diff
@@ -1048,7 +1048,19 @@ def _generate_recovery_conf(
 
         # Writes recovery target
         if target_time:
-            recovery_conf_lines.append("recovery_target_time = '%s'" % target_time)
+            # 'target_time' is the value as it came from '--target-time' command-line
+            # option, which may be without a time zone. When writing the actual Postgres
+            # configuration we should use a value with an explicit time zone set, so we
+            # avoid hitting pitfalls. We use the 'target_datetime' which was prevously
+            # added to 'recovery_info'. It already handles the cases where the user
+            # specifies no time zone, and uses the Barman host time zone as a fallback.
+            # In short: if 'target_time' is present it means the user asked for a
+            # specific point in time, but we need a sanitized value to use in the
+            # Postgres configuration, so we use 'target_datetime'.
+            # See '_set_pitr_targets'.
+            recovery_conf_lines.append(
```

Comment:
We should emit a warning when user doesn't provide a TZ and barman appends it to the recovery time

Reply:
Added that and squashed the commits.